tests/test_misc.py

```diff
@@ -99,3 +99,33 @@ def stats_cb_check_no_brokers(stats_json_str):
 
     global seen_stats_cb_check_no_brokers
     assert seen_stats_cb_check_no_brokers
+
+
+def throttle_cb_instantiate_fail():
+    """ Ensure noncallables raise TypeError"""
+    conf = {
+        'throttle_cb': 1
+    }
+
+    try:
+        confluent_kafka.Producer(conf)
+    except(TypeError) as e:
```

Comment:
`except TypeError as e`

But you should use:
```
  pytest.raises(TypeError):
     confluent_kafka.Producer({'throttle_cb': 1})
```
to have pytest handle this for you